ethstats/ethstats.go

```diff
@@ -132,6 +133,46 @@ func (s *Service) loop() {
 	txSub := emux.Subscribe(core.TxPreEvent{})
 	defer txSub.Unsubscribe()
 
+	// Start a goroutine that exhausts the subsciptions to avoid events piling up
+	var (
+		quitCh = make(chan struct{})
+		headCh = make(chan *types.Block, 1)
+		txCh   = make(chan struct{}, 1)
+	)
+	go func() {
+		var lastTx mclock.AbsTime
+
+		for {
+			select {
+			// Notify of chain head events, but drop if too frequent
```

Comment:
It does drop in 

```
select {
case headCh <- head.Data.(core.ChainHeadEvent).Block:
default:
}
```

If the channel already contains a head, then I'm not waiting to add another one.

The reason I've added an additional 1sec throttling for txs is because the payload is tiny, so when a lot of txs are added, still a lot of requests manage to slip through to ethstats.